### DISCUSSION TITLE
Ignoring Changes from pulumi refresh

### DISCUSSION THREAD

**Discussion by VladAmos** [pulumi/pulumi#10108] [Pulumi in Practice]
Hello,

I learned about `ignoreChanges` recently and it has been very helpful in making sure we set good defaults. I was wondering if there is something like it that works in the opposite direction. My example is a Cloud Run service, however the deployed image changes based on the tag of the image. In that case I want to ignore the change when I run `pulumi refresh` against it the GCP resource. Is there a property that is able to do that?

**Comment by XBeg9** [comment]
same question here, I want `pulumi refresh` to take `ignoreChanges` in count, but currently it's not.

**Comment by blakegong** [comment]
Took me a bit of research. I believe the closest answer to this question can be found [here](https://github.com/pulumi/pulumi-awsx/issues/856#issuecomment-1134772542).

Basically `pulumi refresh` is an action to ONLY pull data from the cloud into your local states, without any drift correction actions. If drift correction is needed, you'd do `pulumi up --refresh` which is basically running 2 steps together:

1. `pulumi refresh` (**read-only** action through the provider: pulls states from cloud to local)
3. `pulumi up` (write action through the provider: compares the updated local states against your Pulumi code, figures out the diff, and corrects the diff through provider APIs)

Since your local Pulumi states would always store the `tag` of your image, saying `ignoreChanges` shouldn't change that during a `pulumi refresh` run, as keeping track of state changes is important in case `ignoreChanges` config is updated later (e.g. `tag` is no longer ignored). Therefore, `pulumi refresh` would faithfully check and tell you "the `tag` value is now different in the cloud and that should be reflected in your states". Notice that this step is purely using **read-only** cloud provider APIs, which doesn't change your deployed resources at all. It does change your stack states though.

The cloud resource changes (aka. drift correction) are done through `pulumi up` when needed, which does respect `ignoreChanges`. That's why `ignoreChanges` only works with `pulumi up` but not `pulumi refresh`.
